src/unix/newlib/mod.rs

```diff
@@ -10,30 +10,58 @@ cfg_if! {
 }
 
 cfg_if! {
-    if #[cfg(any(target_os = "espidf"))] {
+    if #[cfg(target_os = "horizon")] {
+        pub type dev_t = u32;
+        pub type off_t = i64;
+    } else {
         pub type dev_t = ::c_short;
-        pub type ino_t = ::c_ushort;
         pub type off_t = ::c_long;
-    } else if #[cfg(any(target_os = "vita"))] {
-        pub type dev_t = ::c_short;
-        pub type ino_t = ::c_ushort;
-        pub type off_t = ::c_int;
-    } else {
-        pub type dev_t = u32;
+    }
+}
+
+cfg_if! {
+    if #[cfg(target_os = "horizon")] {
         pub type ino_t = u32;
-        pub type off_t = i64;
+    } else if #[cfg(target_arch = "sparc")] {
+        pub type ino_t = ::c_ulong;
+    } else {
+        pub type ino_t = ::c_ushort;
     }
 }
 
 pub type fsblkcnt_t = u64;
 pub type fsfilcnt_t = u32;
 pub type id_t = u32;
-pub type key_t = ::c_int;
+
+cfg_if! {
```

Comment:
For ESP-IDF (and possibly others), you are changing `key_t` from `c_int` to `c_long`. Is this on purpose?

Reply:
https://sourceware.org/git/?p=newlib-cygwin.git;a=blob;f=newlib/libc/include/sys/_types.h;h=d1112d5d102ca509331b9db5b27934b7cc403b32;hb=9e09d6ed83cce4777a5950412647ccc603040409#l108

Reply:
OK, I agree.